db/changes.go

```diff
@@ -241,9 +241,20 @@ func (db *DatabaseCollectionWithUser) buildRevokedFeed(ctx context.Context, ch c
 				// If its less than we can send a standard revocation with Sequence ID as above.
 				// Otherwise: we need to determine whether a previous revision of the document was in the channel prior
 				// to the since value, and only send a revocation if that was the case
+
 				if logEntry.Sequence > sinceVal {
-					requiresRevocation, err := db.wasDocInChannelPriorToRevocation(ctx, logEntry.DocID, singleChannelCache.ChannelID().Name, revocationSinceSeq)
+					// Get doc sync data so we can verify the docs grant history
+					syncData, err := db.GetDocSyncData(ctx, logEntry.DocID)
 					if err != nil {
+						base.InfofCtx(ctx, base.KeyChanges, "Error getting sync data, seq: %v in channel %s, skipping document. Error: %v", seqID, base.UD(singleChannelCache.ChannelID()), err)
+						continue
+					}
+					requiresRevocation, err := db.wasDocInChannelPriorToRevocation(ctx, syncData, logEntry.DocID, singleChannelCache.ChannelID().Name, revocationSinceSeq)
+					if err != nil {
+						if syncData.History == nil {
+							base.TracefCtx(ctx, base.KeyChanges, "Document history empty: %s, no revocation sent", base.UD(logEntry.DocID))
+							continue
+						}
```

Comment:
This seems like it could be done before calling `db.wasDocInChannelPriorToRevocation`?

Reply:
fixed in https://github.com/couchbase/sync_gateway/pull/5930/commits/144083e56a7d20c7a31ac30273957ea46cfc5145